### PR TITLE
refactor(runtime/permissions): use concrete error types

### PR DESCRIPTION
**Pull request by crowlKats**

### CODE REVIEW

File: cli/args/flags.rs
Comment (bartlomieju) [inline cli/args/flags.rs:11055]: Remove?

**Review by bartlomieju — COMMENTED** [review]
Looks okay to me, I'd like to get second pair of eyes on it before we land. @dsherret could you please take a look?

File: cli/args/flags.rs
Comment (crowlKats) [reply cli/args/flags.rs:11055]:
```suggestion
      r.unwrap_err().to_string(),
```

File: ext/websocket/lib.rs
Comment (dsherret) [inline ext/websocket/lib.rs:163]: Oh, this shouldn't be parsing when we have full permissions (I'll look into this later)

File: runtime/permissions/prompter.rs
Comment (dsherret) [inline runtime/permissions/prompter.rs:129]: Might as well just make this an io error then have this return `std::io::Error`?

File: runtime/permissions/prompter.rs
Comment (dsherret) [inline runtime/permissions/prompter.rs:210]: Recommend returning `std::io::Error` here.

**Review by dsherret — APPROVED** [review]
LGTM other than PrompterError.

**Pull request merged**
